pandas/core/dtypes/dtypes.py

```diff
@@ -1216,8 +1254,8 @@ def __setstate__(self, state):
         # pickle -> need to set the settable private ones here (see GH26067)
         self._subtype = state["subtype"]
 
-        # backward-compat older pickles won't have "closed" key
-        self._closed = state.pop("closed", None)
+        # backward-compat older pickles won't have "inclusive" key
+        self._closed = state.pop("inclusive", None)
```

Comment:
i don't think this is tested properly. this should *still* be closed (as that's whats in the older pickles).

Reply:
Have add test in `pandas/tests/dtypes/test_dtypes.py`

```
def test_interval_dtype_with_pickle(self):
        i = pd.IntervalDtype(subtype="int64", inclusive="right")
        dumps_i = pickle.dumps(i)
        loads_i = pickle.loads(dumps_i)

        assert loads_i.subtype == np.dtype("int64")
        assert loads_i.inclusive == "right"
        assert is_interval_dtype(loads_i)
```

Reply:
@jreback After I create  pickle run with version 1.4.2. It generate this pickle file '/home/open_source/pandas_2/pandas/pandas/tests/io/data/legacy_pickle/1.4.2/1.4.2_x86_64_linux_3.9.7.pickle'

Using pandas version environment ''1.5.0.dev0+794.g4f92db3b5d' , 
'''
path = '/home/open_source/pandas_2/pandas/pandas/tests/io/data/legacy_pickle/1.4.2/1.4.2_x86_64_linux_3.9.7.pickle'
data = pd.read_pickle(path)
'''
This error is raised 
```
Traceback (most recent call last):
  File "/home/open_source/pandas_2/pandas/pandas/io/pickle.py", line 205, in read_pickle
    return pickle.load(handles.handle)
  File "pandas/_libs/tslibs/timestamps.pyx", line 151, in pandas._libs.tslibs.timestamps._unpickle_timestamp
TypeError: _unpickle_timestamp() takes exactly 4 positional arguments (3 given)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/open_source/pandas_2/pandas/pandas/compat/pickle_compat.py", line 39, in load_reduce
    stack[-1] = func(*args)
  File "pandas/_libs/tslibs/timestamps.pyx", line 151, in pandas._libs.tslibs.timestamps._unpickle_timestamp
TypeError: _unpickle_timestamp() takes exactly 4 positional arguments (3 given)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/open_source/pandas_2/development/development_5.py", line 36, in <module>
    data = pd.read_pickle(path)
  File "/home/open_source/pandas_2/pandas/pandas/io/pickle.py", line 210, in read_pickle
    return pc.load(handles.handle, encoding=None)
  File "/home/open_source/pandas_2/pandas/pandas/compat/pickle_compat.py", line 272, in load
    return up.load()
  File "/root/anaconda3/lib/python3.9/pickle.py", line 1212, in load
    dispatch[key[0]](self)
  File "/home/open_source/pandas_2/pandas/pandas/compat/pickle_compat.py", line 60, in load_reduce
    elif args and issubclass(args[0], PeriodArray):
TypeError: issubclass() arg 1 must be a class
```

Reply:
> TypeError: _unpickle_timestamp() takes exactly 4 positional arguments (3 given)

so we have some incompatibilites which this is picking up. can create a separate issue to fix that or try here. (basically this function needs to be made backwards compatible).

cc @jbrockmendel

Reply:
looks like we need to set the default value of reso in _unpickle_timestamp to NPY_FR_ns

Reply:
Pickle test already done from this commit https://github.com/pandas-dev/pandas/commit/1ce21664848a1128ea1456d4fd24d76f045a74df